source/lac/psblas_vector.cc

```diff
@@ -0,0 +1,681 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2019 - 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+#include "deal.II/base/exception_macros.h"
+#include "deal.II/base/exceptions.h"
+#include <deal.II/base/memory_consumption.h>
+#include <deal.II/base/mpi.h>
+
+#include <deal.II/lac/psblas_vector.h>
+
+#ifdef DEAL_II_WITH_PSBLAS
+#  include <psb_base_cbind.h>
+#  include <psb_c_base.h>
+#  include <psb_c_dbase.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace PSCToolkit
+{
+
+  Vector::Vector()
+    : psblas_vector(nullptr)
+    , psblas_context(InitFinalize::get_psblas_context())
+    , psblas_descriptor(nullptr)
+    , ghosted(false)
+    , state(internal::State::Default)
+    , last_action(VectorOperation::unknown)
+  {}
+
+
+  Vector::Vector(const Vector &v)
+    : Vector::Vector()
+  {
+    if (v.has_ghost_elements())
+      reinit(v.owned_elements, v.ghost_indices, v.communicator);
+    else
+      reinit(v.owned_elements, v.communicator);
+
+    this->operator=(v);
+  }
+
+
+  Vector::~Vector()
+  {
+    if (state != internal::State::Default)
+      {
+        int err;
```

Comment:
```suggestion
        int ierr;
```
we usually use `ierr` in the library for an integer error code.